tests/baselines/reference/controlFlowAliasing.errors.txt

```diff
@@ -304,9 +297,6 @@ tests/cases/conformance/controlFlow/controlFlowAliasing.ts(280,5): error TS2454:
                 // Some narrowings may be invalidated due to later assignments.
                 let s: string;
                 s = this.x;
-                ~

```

Comment:
I would expect this to match `s  = x` on the next line, as it does in the previous diff. Why does the error go away for `s = this.x` here, but not for `s = x`?
For that matter, why does assignment in the `else` branch prevent narrowing in the `if` branch?

Reply:
There's an assignment to `x` in the `else` branch that causes `x` to viewed as mutable and therefore it isn't narrowed. There's a comment to that effect in the test, so not unexpected.